src/features/student/components/badges/StudentLevelBadge/StudentLevelBadge.vue

```diff
@@ -1,47 +1,49 @@
 <script setup lang="ts">
-import { type LevelDTO, type LevelDTO_Temp, LevelStatus } from '@/types'
+import type { SkillLevelViewDTO } from '@/types'
+import { type SkillLevelOverviewDTO, SkillLevelStatus } from '@/api/avenir-esr'
 import { AvBadge } from '@/ui'
 import { useI18n } from 'vue-i18n'
 
-const { level } = defineProps<{ level: LevelDTO | LevelDTO_Temp }>()
+const { level } = defineProps<{ level: SkillLevelOverviewDTO | SkillLevelViewDTO }>()
 
 const { t } = useI18n()
 
-function levelToBadge (level: LevelDTO | LevelDTO_Temp) {
-  switch (level.status) {
-    case LevelStatus.NOT_VALIDATED:
-      return {
-        labelkey: 'student.badges.studentLevelBadge.status.notValidated',
-        color: 'var(--light-foreground-error)',
-        backgroundColor: 'var(--light-background-error)',
-        iconPath: '/assets/icons/close-circle-outline.svg'
-      }
-    case LevelStatus.TO_EVALUATE:
+function levelToBadge (status: SkillLevelStatus) {
+  switch (status) {
+    case SkillLevelStatus.NOT_STARTED:
+    case SkillLevelStatus.TO_BE_EVALUATED:
       return {
         labelkey: 'student.badges.studentLevelBadge.status.inProgress',
         color: 'var(--dark-background-primary1)',
         backgroundColor: 'var(--light-background-primary2)',
         iconPath: '/assets/icons/hourglass.svg'
       }
-    case LevelStatus.UNDER_REVIEW:
+    case SkillLevelStatus.UNDER_REVIEW:
       return {
         labelkey: 'student.badges.studentLevelBadge.status.underReview',
         color: 'var(--dark-background-critical)',
         backgroundColor: 'var(--light-background-critical)',
         iconPath: '/assets/icons/dots-horizontal-circle-outline.svg'
       }
-    case LevelStatus.VALIDATED:
+    case SkillLevelStatus.VALIDATED:
       return {
         labelkey: 'student.badges.studentLevelBadge.status.validated',
         color: 'var(--light-foreground-success)',
         backgroundColor: 'var(--light-background-success)',
         iconPath: '/assets/icons/check-circle.svg'
       }
+    case SkillLevelStatus.FAILED:
+      return {
+        labelkey: 'student.badges.studentLevelBadge.status.failed',
+        color: 'var(--light-foreground-error)',
+        backgroundColor: 'var(--light-background-error)',
+        iconPath: '/assets/icons/close-circle-outline.svg'
+      }
   }
 }
 
-const levelBadge = computed(() => levelToBadge(level))
-const label = computed(() => t(levelBadge.value.labelkey))
+const levelBadge = computed(() => levelToBadge(level.status))
+const label = computed(() => t(levelBadge.value?.labelkey))
```

Comment:
```suggestion
const label = computed(() => t(levelBadge.value.labelkey))
```

Une donnée est forcément retournée, pas besoin du `?` normalement

Reply:
@p-beger je l'ai mis lorsque j'ai fait un changes et je l'ai revert, mais je l'ai oublié, bien vue